hieradata/clients/controller.cert.ci.jenkins.io.yaml

```
@@ -92,8 +92,10 @@ profile::jenkinscontroller::jcasc:
         url: "https://groovy.jfrog.io/artifactory/dist-release-local/groovy-zips/apache-groovy-binary-3.0.6.zip"
   cloud_agents:
     azure-vm-agents:
-      azureCredentialsId: "azure-sp-agents" # Managed manually
-      resource_group: "cert-ci-jenkins-io-ephemeral-agents"
+      azureVMClouds:
```

Out of curiosity, what the reasoning behind the choice of camelCase here?

---

As I'm not really caring for casing conventions, It comes from me writing things without even noticing.

WDYT about?

```suggestion
      azure-vm-clouds:
```

---

```suggestion
      azure-cloud-vms:
```

---

The name of the cloud is "Azure VM" and its JCasc key is `azureVM` because it is the https://plugins.jenkins.io/azure-vm-agents/ plugin.

As such, the `azure-cloud-vms` is invalid semantically.


WDYT about `clouds` since this is a sub-key of `azure-vm-agents` ?

---

Updated the "casing" for the scope of this PR: WDYT?

---

New updates following our discussions:

- Case changed only on the scope of this PR (the file deserves more of it but it is non functionnal)
- CamelCase kept for attributres directly mapped from hieradata to JCasc in the template
- snake_case for hieradata pure attribiutes
- Kept a few kebab-case when YAML key is mapped to a functionnal field (such as the agent cloud name `azure-vms`)